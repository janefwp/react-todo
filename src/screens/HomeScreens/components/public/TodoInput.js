import React from 'react'
import './Todocom.scss'
function TodoInput(props) {

    return (
        <fieldset>
                <label>{props.label}</label>
                <props.as  name={props.name} required={props.required} onChange={props.onChange}></props.as>          
        </fieldset>
    )
}

export default TodoInput
