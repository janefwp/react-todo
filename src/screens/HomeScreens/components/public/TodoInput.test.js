import React from 'react';
import renderer from 'react-test-renderer';
import TodoInput from './TodoInput'

it('renders correctly when there is input type', () => {
    const tree = renderer.create(<TodoInput as="input"/>).toJSON();
    expect(tree).toMatchSnapshot();
  });

it('renders correctly when there is textarea type', () => {
    const tree = renderer.create(<TodoInput as="textarea"/>).toJSON();
    expect(tree).toMatchSnapshot();
  });