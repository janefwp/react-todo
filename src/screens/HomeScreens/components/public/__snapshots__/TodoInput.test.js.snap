// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when there is input type 1`] = `
<fieldset>
  <label />
  <input />
</fieldset>
`;

exports[`renders correctly when there is input type 2`] = `
<fieldset>
  <label />
  <textarea />
</fieldset>
`;

exports[`renders correctly when there is textarea type 1`] = `
<fieldset>
  <label />
  <textarea />
</fieldset>
`;
